grails-app/services/org/pih/warehouse/core/LocationService.groovy

```diff
@@ -718,4 +724,83 @@ class LocationService {
         }
         existingLocation.delete(flush: true)
     }
+
+    Location createOrUpdateLocation(Map params) {
+        Location location
+        if (params.id) {
+            location = Location.findById(params.id)
+        } else if (params.locationNumber) {
+            location = Location.findByNameOrLocationNumber(params.name, params.locationNumber)
+        } else {
+            location = Location.findByName(params.name)
+        }
+        if (!location) {
+            location = new Location()
+        }
+
+        location.name = params.name
+        location.active = Boolean.valueOf(params.active)
+        location.locationNumber = params.locationNumber
+        location.locationGroup = params.locationGroup ? LocationGroup.findByName(params.locationGroup) : null
+        location.parentLocation = params.parentLocation ? Location.findByNameOrLocationNumber(params.parentLocation, params.parentLocation) : null
+        location.organization = params.organization ? Organization.findByCodeOrName(params.organization, params.organization) : null
+        location.address = createOrUpdateAddress(params, location?.address?.id)
+        if (!location.inventory && !location.parentLocation) {
+            location.inventory = new Inventory(['warehouse': location])
+        }
+        def locationType = null
+        if (params.locationType) {
+            String locationTypeName = LocalizationUtil.getDefaultString(params.locationType as String)
+            // TODO: Replace with a single GORM .find with Closure when in Grails 3 (available since Grails 2.0)
```

Comment:
as far as I remember I was the one to add this TODO - could you try to proceed it? It should be as simple as removing the `.findAllByNameLike` here - as this find would be then the GORM one, not the one related to List interface.